sh/abandon_chain.sh

```diff
@@ -0,0 +1,249 @@
+IFS='' read -p 'What address will you submit with?: ' -e -r -i 0xEf37aD2BACD70119F141140f7B5E46Cd53a65fc4 signer
+declare -r signer
+
+. "$project_root"/sh/common_safe_deployer.sh
+. "$project_root"/sh/common_wallet_type.sh
+
+# Verify the initial owner is not already the only owner
+if (( ${#owners_array[@]} == 1 )) && [[ $(cast to-checksum "${owners_array[0]}") = $(cast to-checksum "$initial_owner") ]] ; then
+    echo 'Safe is already abandoned (initial owner is the only owner)' >&2
+    exit 0
+fi
+
+declare -r removeOwner_sig='removeOwner(address,address,uint256)'
+declare -r swapOwner_sig='swapOwner(address,address,address)'
+declare -r sentinel='0x0000000000000000000000000000000000000001'
+
+declare -a calls=()
+
+# Remove owners until only one remains
+declare -a current_owners=("${owners_array[@]}")
+
+while (( ${#current_owners[@]} > 1 )) ; do
+    declare owner_to_remove="${current_owners[0]}"
+
+    declare removeOwner_call
+    removeOwner_call="$(cast calldata "$removeOwner_sig" "$sentinel" "$owner_to_remove" 1)"
+
+    calls+=(
+        "$(
+            cast concat-hex                                              \
+            0x00                                                         \
+            "$safe_address"                                              \
+            "$(cast to-uint256 0)"                                       \
+            "$(cast to-uint256 "$(((${#removeOwner_call} - 2) / 2))")"   \
+            "$removeOwner_call"
+        )"
+    )
+
+    # Update working state
+    current_owners=("${current_owners[@]:1}")
+done
+
+# Swap the last remaining owner with initial_owner
+declare last_owner="${current_owners[0]}"
+
+declare swapOwner_call
+swapOwner_call="$(cast calldata "$swapOwner_sig" "$sentinel" "$last_owner" "$initial_owner")"
+
+calls+=(
+    "$(
+        cast concat-hex                                            \
+        0x00                                                       \
+        "$safe_address"                                            \
+        "$(cast to-uint256 0)"                                     \
+        "$(cast to-uint256 "$(((${#swapOwner_call} - 2) / 2))")"   \
+        "$swapOwner_call"
+    )"
+)
+
+# Wrap in multiSend call
+declare multisend_calldata
+multisend_calldata="$(cast calldata "$multisend_sig" "$(cast concat-hex "${calls[@]}")")"
+declare -r multisend_calldata
+
+# set minimum gas price (mostly for Arbitrum and BNB)
+declare -i min_gas_price
+min_gas_price="$(get_config minGasPriceGwei)"
+min_gas_price=$((min_gas_price * 1000000000))
+declare -r -i min_gas_price
+declare -i gas_price
+gas_price="$(cast gas-price --rpc-url "$rpc_url")"
+if (( gas_price < min_gas_price )) ; then
+    echo 'Setting gas price to minimum of '$((min_gas_price / 1000000000))' gwei' >&2
+    gas_price=$min_gas_price
+fi
+declare -r -i gas_price
+declare -i gas_estimate_multiplier
+gas_estimate_multiplier="$(get_config gasMultiplierPercent)"
+declare -r -i gas_estimate_multiplier
```

Comment:
Can you reify/DRY this with the other scripts and put it in one of the `common_*.sh` files?